### ISSUE TITLE
PowerShellGallery downloads show invalid

### ISSUE THREAD

**Issue by thedavecarroll** [badges/shields#2078]
After today's release of the updated PowerShellGallery, the download data shows invalid, even on the shields.io site.

**Comment by thedavecarroll** [comment]
The version badge is also showing invalid.

**Comment by chris48s** [comment]
Hi. thanks for reporting this.

It looks like the issue here is that since the recent changes to powershell gallery (related reading: https://blogs.msdn.microsoft.com/powershell/2018/09/12/new-look-and-features-for-powershell-gallery/ ), the API no longer serves JSON when requested. For example

```
$ curl -H "Accept: application/atom+json,application/json" "https://www.powershellgallery.com/api/v2/Packages()?$filter=Id%20eq%20'ACMESharp'%20and%20IsLatestVersion%20eq%20true"
```

now returns an XML response.

This does not affect resharper or chocolatey which share the same NuGet v2 API parsing code.

I've raised an issue over on the powershell gallery support repo - https://github.com/PowerShell/PowerShellGallery/issues/21 Follow there for updates/progress.

If we're not able to get the data as JSON, the badge code would need to be rewritten to parse the XML feed instead, but hopefully there is a quicker solution.

**Comment by thedavecarroll** [comment]
Thank you so much for responding. It's given me a lot of insight into parsing data from the gallery. So much easier than screen scraping. lol

**Comment by cormacpayne** [comment]
+1 on behalf of [Azure PowerShell](https://github.com/Azure/azure-powershell) 😀

**Comment by iricigor** [comment]
As per comment on https://github.com/PowerShell/PowerShellGallery/issues/21 there is a link available that gives JSON back. How complicated is to change  `www.powershellgallery.com` to `msconfiggallery.cloudapp.net`? As I see the only definition is [here](https://github.com/badges/shields/blob/ca6909938920c55872e792656c5d903699346527/services/nuget/nuget.service.js#L347).

`curl -H "Accept: application/atom+json,application/json" "https://msconfiggallery.cloudapp.net/api/v2/Packages()?$filter=Id%20eq%20'ACMESharp'%20and%20IsLatestVersion%20eq%20true"`

**Comment by chris48s** [comment]
Yeah that issue seems to have gone quiet and the "temporary workaround" is still available. I've opened PR #2189 proposing the change.

**Comment by jzabroski** [comment]
The PowerShell team is famous for not testing their software, in my experience...

**Comment by paulmelnikow** [comment]
I've proposed a fix for this in #2257.

Could anyone who's having this problem confirms this fixes it?

http://shields-staging-pr-2257.herokuapp.com/

**Comment by iricigor** [comment]
Yes, these new badges looks OK

Here is page with new badges - properly working

https://github.com/iricigor/FIFA2018/tree/new-test-badges

![image](https://user-images.githubusercontent.com/15923171/48430489-c195b480-e76f-11e8-94b0-f679dc0245ec.png)


Here is old version of the page - non working

https://github.com/iricigor/FIFA2018

![image](https://user-images.githubusercontent.com/15923171/48430541-d70ade80-e76f-11e8-9448-2fb3b9f13ba6.png)

**Comment by jzabroski** [comment]
Seems to be broken again.

**Comment by calebcartwright** [comment]
@jzabroski do you have badge URLs/package names you can share? I checked with a handful of the packages referenced in this thread and they appear to be working for me 

![](https://img.shields.io/powershellgallery/v/FIFA2018.svg)
![](https://img.shields.io/powershellgallery/dt/FIFA2018.svg)

**Comment by jzabroski** [comment]
@iricigor example here: https://github.com/iricigor/FIFA2018/tree/new-test-badges

**Comment by chris48s** [comment]
The badges URLs in the README you've linked to aren't requesting from shields.io


Change

http://shields-staging-pr-2257.herokuapp.com/powershellgallery/v/FIFA2018.svg?label=latest+version and http://shields-staging-pr-2257.herokuapp.com/powershellgallery/dt/FIFA2018.svg?label=downloads

to

https://img.shields.io/powershellgallery/v/FIFA2018.svg?label=latest+version and https://img.shields.io/powershellgallery/dt/FIFA2018.svg?label=downloads

to fix

**Comment by calebcartwright** [comment]
The badges in the linked example are not pointed to Shields.io, they are pointed to a non-existent staging app ((http://shields-staging-pr-2257.herokuapp.com)

https://raw.githubusercontent.com/iricigor/FIFA2018/new-test-badges/README.md
http://shields-staging-pr-2257.herokuapp.com/powershellgallery/v/FIFA2018.svg
http://shields-staging-pr-2257.herokuapp.com/powershellgallery/dt/FIFA2018.svg

[![latest version](http://shields-staging-pr-2257.herokuapp.com/powershellgallery/v/FIFA2018.svg?label=latest+version)](https://www.powershellgallery.com/packages/FIFA2018)
[![downloads](http://shields-staging-pr-2257.herokuapp.com/powershellgallery/dt/FIFA2018.svg?label=downloads)](https://www.powershellgallery.com/packages/FIFA2018)

**Comment by chris48s** [comment]
snap :clubs: :clubs: :smile:

**Comment by calebcartwright** [comment]
Lol @chris48s  you beat me to it!

**Comment by jzabroski** [comment]
Sorry for the distraction guys.  I was using this GitHub issue to find a working example for a module I was planning to publish.

**Comment by calebcartwright** [comment]
No worries! There's a new experience in the UI (at least I think it was rolled out post this thread) that simplifies building badges too
